lib/db.h

```diff
@@ -36,21 +36,24 @@
 #define _FRR_DB_H_
 #ifdef HAVE_SQLITE3
 
+#include "compiler.h"
 #include <sqlite3.h>
 
 #ifdef __cplusplus
 extern "C" {
 #endif
 
-extern int db_init(const char *path_fmt, ...);
+extern int db_init(const char *path_fmt, ...) PRINTFRR(1, 2);
 extern int db_close(void);
+/* WARNING: sqlite format string! not printf compatible! */
 extern int db_bindf(struct sqlite3_stmt *ss, const char *fmt, ...);
 extern struct sqlite3_stmt *db_prepare_len(const char *stmt, int stmtlen);
 extern struct sqlite3_stmt *db_prepare(const char *stmt);
 extern int db_run(struct sqlite3_stmt *ss);
+/* WARNING: sqlite format string! not printf compatible! */
```

Comment:
I think this should read: ``not scanf compatible`` rather than ``not printf compatible``?

Reply:
Makes sense, edited.